File: app/src/main/java/ru/job4j/todokt/db/TaskDao.java
package ru.job4j.todokt.db;





import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import io.reactivex.Flowable;
import io.reactivex.Single;


/**
 * @author Dmitry Kolganov (mailto:dmk78@inbox.ru)
 * @version $Id$
 * @since 12.02.2020
 */

@Dao
public interface TaskDao {


    @Query("SELECT * FROM tasks")
    Flowable<List<Task>> getAllTasks();

    @Query("SELECT * FROM tasks WHERE id == :id")
    Single<Task> getTaskById(int id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTask(Task task);

    @Delete
    void deleteTask(Task task);

    @Update
    void updateTask(Task task);

    @Query("DELETE FROM tasks")
    void deleteAllTasks();









}

File: app/src/main/java/ru/job4j/todokt/db/TaskDao.kt
package ru.job4j.todokt.db

import androidx.room.*
import io.reactivex.Flowable
import io.reactivex.Single

/**
 * @author Dmitry Kolganov (mailto:dmk78@inbox.ru)
 * @version $Id$
 * @since 12.02.2020
 */
@Dao
interface TaskDao {
    @Query("SELECT * FROM tasks")
    fun getAllTasks(): Flowable<List<Task?>?>?

    @Query("SELECT * FROM tasks WHERE id == :id")
    fun getTaskById(id: Int): Single<Task?>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTask(task: Task?)

    @Delete
    fun deleteTask(task: Task?)

    @Update
    fun updateTask(task: Task?)

    @Query("DELETE FROM tasks")
    fun deleteAllTasks()
}